persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/ClinicalDataSparkRepository.java

```diff
@@ -0,0 +1,289 @@
+
+    @Value("${data.parquet.folder}")
+    private String PARQUET_DIR;
+
+    private static Log logger = LogFactory.getLog(ClinicalDataSparkRepository.class);
+    private static final String MUTATION_COUNT = "MUTATION_COUNT";
+    private static final String FRACTION_GENOME_ALTERED = "FRACTION_GENOME_ALTERED";
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataOfSampleInStudy(String studyId, String sampleId, String attributeId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaSampleClinicalData(String studyId, String sampleId, String attributeId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataOfPatientInStudy(String studyId, String patientId, String attributeId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaPatientClinicalData(String studyId, String patientId, String attributeId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataInStudy(String studyId, String attributeId, String clinicalDataType, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaAllClinicalData(String studyId, String attributeId, String clinicalDataType) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> fetchAllClinicalDataInStudy(String studyId, List<String> ids, List<String> attributeIds, String clinicalDataType, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaClinicalDataInStudy(String studyId, List<String> ids, List<String> attributeIds, String clinicalDataType) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> fetchClinicalData(List<String> studyIds, List<String> ids, List<String> attributeIds,
+                                                String clinicalDataType, String projection) {
+        List<Dataset<Row>> res = new ArrayList<>();
+        
+        Set<String> studyIdSet = new HashSet<>(studyIds);
+        String dataFile = ParquetConstants.CLINICAL_PATIENT;
+        if (PersistenceConstants.SAMPLE_CLINICAL_DATA_TYPE.equalsIgnoreCase(clinicalDataType)) {
+            dataFile = ParquetConstants.CLINICAL_SAMPLE;
+        }
+        Dataset<Row> clinicalData = loadStudyFiles(studyIdSet, dataFile);
+        
+        // join with clinical sample data to get sample id information.
+        if ("PATIENT".equalsIgnoreCase(clinicalDataType)) {
```

Comment:
create `PATIENT` constant similar to `SAMPLE_CLINICAL_DATA_TYPE`

Reply:
done